fl4health/clients/fenda_client.py

```diff
@@ -1,12 +1,229 @@
-from flwr.common.typing import Config
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Tuple
 
+import torch
+from flwr.common.typing import Config, NDArrays
+
+from fl4health.checkpointing.checkpointer import TorchCheckpointer
 from fl4health.clients.basic_client import BasicClient
 from fl4health.model_bases.fenda_base import FendaModel
 from fl4health.parameter_exchange.layer_exchanger import FixedLayerExchanger
 from fl4health.parameter_exchange.parameter_exchanger_base import ParameterExchanger
+from fl4health.utils.losses import Losses, LossMeterType
+from fl4health.utils.metrics import Metric, MetricMeterType
 
 
 class FendaClient(BasicClient):
+    def __init__(
+        self,
+        data_path: Path,
+        metrics: Sequence[Metric],
+        device: torch.device,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        metric_meter_type: MetricMeterType = MetricMeterType.AVERAGE,
+        checkpointer: Optional[TorchCheckpointer] = None,
+        temperature: Optional[float] = 0.5,
+        perFCL_loss_weights: Optional[Tuple[float, float]] = (0.0, 0.0),
+        cos_sim_loss_weight: Optional[float] = 0.0,
+        contrastive_loss_weight: Optional[float] = 0.0,
+    ) -> None:
+        super().__init__(
+            data_path=data_path,
+            metrics=metrics,
+            device=device,
+            loss_meter_type=loss_meter_type,
+            metric_meter_type=metric_meter_type,
+            checkpointer=checkpointer,
+        )
+        """This module is used to init fenda client with various auxiliary loss functions.
+        These losses will be activated only when their weights are not 0.0.
+        Args:
+            data_path: Path to the data directory.
+            metrics: List of metrics to be used for evaluation.
+            device: Device to be used for training.
+            loss_meter_type: Type of loss meter to be used.
+            metric_meter_type: Type of metric meter to be used.
+            checkpointer: Checkpointer to be used for checkpointing.
+            temperature: Temperature to be used for contrastive loss.
+            perFCL_loss_weights: Weights to be used for perFCL loss.
+            Each value associate with one of two contrastive losses in perFCL loss.
+            cos_sim_loss_weight: Weight to be used for cosine similarity loss.
+            contrastive_loss_weight: Weight to be used for contrastive loss.
+        """
+        self.perFCL_loss_weights = perFCL_loss_weights
```

Comment:
This is sort of ambiguous in the python standards, but we've unified on having acronyms be lower case when they are variables and only the first letter by capitalized if they are classes.

That is, `perFCL_loss_weights` -> `perfcl_loss_weights` or `per_fcl_loss_weights` as a variable and `PerFclClass` if it were a class object.